cmd/agent/app/agent.go

```diff
@@ -223,6 +224,8 @@ func startClusterStatusController(ctx controllerscontext.Context) (bool, error)
 		ClusterStatusUpdateFrequency:      ctx.Opts.ClusterStatusUpdateFrequency,
 		ClusterLeaseDuration:              ctx.Opts.ClusterLeaseDuration,
 		ClusterLeaseRenewIntervalFraction: ctx.Opts.ClusterLeaseRenewIntervalFraction,
+		ClusterFailureThreshold:           ctx.Opts.ClusterFailureThreshold,
+		ClusterFailureCounter:             make(map[string]int),
```

Comment:
We can initialize it inside the controller. See here: https://github.com/karmada-io/karmada/pull/1781#discussion_r874274821